File: developers/academy/py/starter_text_data/_snippets/102_collection.py
# CreateMovieCollection
import weaviate

# CreateMovieCollection  # SubmoduleImport
import weaviate.classes.config as wc
import os

# CreateMovieCollection  # END SubmoduleImport

# END CreateMovieCollection
client = weaviate.connect_to_wcs(
    cluster_url=os.getenv("WCS_DEMO_URL"),  # Replace with your WCS URL
    auth_credentials=weaviate.auth.AuthApiKey(
        os.getenv("WCS_DEMO_ADMIN_KEY")
    ),  # Replace with your WCS key
)

# CreateMovieCollection
# Instantiate your client (not shown). e.g.:
# headers = {"X-OpenAI-Api-Key": os.getenv("OPENAI_APIKEY")}  # Replace with your OpenAI API key
# client = weaviate.connect_to_wcs(..., headers=headers) or
# client = weaviate.connect_to_local(..., headers=headers)

# END CreateMovieCollection

# Actual instantiation

client.collections.delete("Movie")

# CreateMovieCollection
client.collections.create(
    name="Movie",
    properties=[
        wc.Property(name="title", data_type=wc.DataType.TEXT),
        wc.Property(name="overview", data_type=wc.DataType.TEXT),
        wc.Property(name="vote_average", data_type=wc.DataType.NUMBER),
        wc.Property(name="genre_ids", data_type=wc.DataType.INT_ARRAY),
        wc.Property(name="release_date", data_type=wc.DataType.DATE),
        wc.Property(name="tmdb_id", data_type=wc.DataType.INT),
    ],
    # Define the vectorizer module
    vectorizer_config=wc.Configure.Vectorizer.text2vec_openai(),
    # Define the generative module
    generative_config=wc.Configure.Generative.openai()
    # END generativeDefinition  # CreateMovieCollection
)

client.close()
# END CreateMovieCollection


# BatchImportData
import weaviate
import pandas as pd
import requests
from datetime import datetime, timezone
import json
from weaviate.util import generate_uuid5
from tqdm import tqdm
import os

# END BatchImportData
headers = {"X-OpenAI-Api-Key": os.getenv("OPENAI_APIKEY")}
client = weaviate.connect_to_wcs(
    cluster_url=os.getenv("WCS_DEMO_URL"),  # Replace with your WCS URL
    auth_credentials=weaviate.auth.AuthApiKey(
        os.getenv("WCS_DEMO_ADMIN_KEY")
    ),  # Replace with your WCS key
    headers=headers,
)

# BatchImportData
# Instantiate your client (not shown). e.g.:
# client = weaviate.connect_to_wcs(...) or
# client = weaviate.connect_to_local(...)

# END BatchImportData

# BatchImportData
data_url = "https://raw.githubusercontent.com/weaviate-tutorials/edu-datasets/main/movies_data_1990_2024.json"
resp = requests.get(data_url)
df = pd.DataFrame(resp.json())

# Get the collection
movies = client.collections.get("Movie")

# Enter context manager
with movies.batch.rate_limit(2400) as batch:
    # Loop through the data
    for i, movie in tqdm(df.iterrows()):
        # Convert data types
        # Convert a JSON date to `datetime` and add time zone information
        release_date = datetime.strptime(movie["release_date"], "%Y-%m-%d").replace(
            tzinfo=timezone.utc
        )
        # Convert a JSON array to a list of integers
        genre_ids = json.loads(movie["genre_ids"])

        # Build the object payload
        movie_obj = {
            "title": movie["title"],
            "overview": movie["overview"],
            "vote_average": movie["vote_average"],
            "genre_ids": genre_ids,
            "release_date": release_date,
            "tmdb_id": movie["id"],
        }

        # Add object to batch queue
        batch.add_object(
            properties=movie_obj,
            uuid=generate_uuid5(movie["id"])
            # references=reference_obj  # You can add references here
        )
        # Batcher automatically sends batches

# Check for failed objects
if len(movies.batch.failed_objects) > 0:
    print(f"Failed to import {len(movies.batch.failed_objects)} objects")

client.close()

File: developers/academy/js/starter_text_data/_snippets/102_collection.ts
// CreateMovieCollection // SubmoduleImport // BatchImportData
import weaviate from "weaviate-client";
// END BatchImportData
import { WeaviateClient, configure, vectorizer } from "weaviate-client";
// END CreateMovieCollection  // END SubmoduleImport


// BatchImportData
import { generateUuid5 } from "weaviate-client";

// END BatchImportData

// BatchImportData
let client: WeaviateClient;
// CreateMovieCollection  // END BatchImportData

// END CreateMovieCollection

client = await weaviate.connectToWeaviateCloud(process.env.WCD_URL as string,{
    authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
  } 
)
// CreateMovieCollection
// Instantiate your client (not shown). e.g.:
// const requestHeaders = {'X-OpenAI-Api-Key': process.env.OPENAI_APIKEY as string,}
// client = weaviate.connectToWeaviateCloud(..., headers: requestHeaders) or
// client = weaviate.connectToLocal(..., headers: requestHeaders)

// END CreateMovieCollection

const requestHeaders = {'X-OpenAI-Api-Key': process.env.OPENAI_APIKEY as string,}


client = await weaviate.connectToWeaviateCloud(
  process.env.WCD_URL as string,
  {
    authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
    headers: requestHeaders
  } 
)

// END CreateMovieCollection

// Actual instantiation

client.collections.delete("Movie")

// CreateMovieCollection
await client.collections.create({
    name: "Movie",
    properties: [
        { name: "title", dataType: configure.dataType.TEXT},
        { name: "overview", dataType: configure.dataType.TEXT},
        { name: "vote_average", dataType: configure.dataType.NUMBER},
        { name: "genre_ids", dataType: configure.dataType.INT_ARRAY},
        { name: "release_date", dataType: configure.dataType.DATE},
        { name: "tmdb_id", dataType: configure.dataType.INT},
    ],
    // Define the vectorizer module
    vectorizers: vectorizer.text2VecOpenAI(),
    // Define the generative module
    generative: configure.generative.openAI(),
    // END generativeDefinition  // CreateMovieCollection
  })

client.close()
// END CreateMovieCollection


client = await weaviate.connectToWeaviateCloud(
  process.env.WCD_URL as string,
  {
    authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
    headers: {
      'X-OpenAI-Api-Key': process.env.OPENAI_APIKEY as string,  // Replace with your inference API key
    }
  } 
)

// BatchImportData

// Instantiate your client (not shown). e.g.:
// client = weaviate.connectToWeaviateCloud(...) or
// client = weaviate.connectToLocal(...)

// END BatchImportData

// BatchImportData
const dataUrl = "https://raw.githubusercontent.com/weaviate-tutorials/edu-datasets/main/movies_data_1990_2024.json"
const response = await fetch(dataUrl)
const data = await response.json()

// Get the collection
const movies = client.collections.get("Movie")

// Set a counter and initialize Weaviate Object
let itemsToInsert: Object[] = []
let counter = 0;

// Iterate through data
for (const key of Object.keys(data['title'])) {
    counter++;
    if(counter % 1000 == 0)
        console.log(`Import: ${counter}`)
    // END Iterate through data // END BatchImportData
    // BatchImportData

    let genreIds: []

    // Format genre_ids and release_date
    const parsedArray = JSON.parse(data['genre_ids'][key]);
    genreIds = parsedArray.map(item => parseInt(item, 10));
    let releaseDate = new Date(data['release_date'][key])

    // Build the object payload
    let movieObject = {
        title: data['title'][key],
        overview: data['overview'][key],
        vote_average: data['vote_average'][key],
        genre_ids: genreIds,
        release_date: releaseDate,
        tmdb_id: data['id'][key],
    }
    // Insert
    let objectToInsert = {
        properties: movieObject,
        uuid: generateUuid5(data['title'][key])
    }

    // Add object to batching array
    itemsToInsert.push(objectToInsert)

    if(itemsToInsert.length == 2000) {
        // Batch insert 2000 items and clear batch array
        const response = await movies.data.insertMany(itemsToInsert)
        itemsToInsert = []
        if(response.hasErrors) {
            throw new Error("Something went wrong in import!")
        }
    }
    // END BatchImportData // END Insert
    // BatchImportData // Iterate through data
    // ... other operations
}
// END Iterate through data // END BatchImportData
// BatchImportData
// insert the remaining objects
if(itemsToInsert.length > 0) {
  // Batch insert any remaining items
  const response = await movies.data.insertMany(itemsToInsert)
  console.log("Done Importing")

  // END BatchImportData
  // Handle Errors // BatchImportData
  if(response.hasErrors) {
      throw new Error("Something went wrong in import!")
  }
  // END BatchImportData // END Handle Errors
  // BatchImportData
}

client.close()
// END BatchImportData
